File: epi_judge_python/primitive_divide.py
from test_framework import generic_test


def divide(x: int, y: int) -> int:
    # TODO - you fill in here.
    return 0


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('primitive_divide.py',
                                       'primitive_divide.tsv', divide))

File: epi_judge_java_solutions/epi/PrimitiveDivide.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class PrimitiveDivide {

  @EpiTest(testDataFile = "primitive_divide.tsv")
  public static int divide(int x, int y) {

    int result = 0;
    int power = 32;
    long yPower = (long)y << power;
    while (x >= y) {
      while (yPower > x) {
        yPower >>>= 1;
        --power;
      }

      result += 1 << power;
      x -= yPower;
    }
    return result;
  }

  public static void main(String[] args) {
    System.exit(
        GenericTest
            .runFromAnnotations(args, "PrimitiveDivide.java",
                                new Object() {}.getClass().getEnclosingClass())
            .ordinal());
  }
}
